webpack/assets/javascripts/react_app/redux/reducers/toasts/index.js

```diff
@@ -0,0 +1,28 @@
+import { TOASTS_ADD, TOASTS_HIDE, TOASTS_DELETE } from '../../consts';
+import Immutable from 'seamless-immutable';
+
+const initialState = Immutable({
+  messages: {}
+});
+
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case TOASTS_ADD: {
+      return state.setIn(['messages', payload.key], payload.message);
+    }
+
+    case TOASTS_HIDE: {
+      return state.setIn(['messages', payload.key, 'visible'], false);
+    }
+
+    case TOASTS_DELETE: {
```

Comment:
I wonder, is there any point in having both hide and delete? after clicking the X there is no way to show the notification anyways, why not unify this? It will also let us simplify the implementation since we don't have to check visibility at all.